metrics/metricstest/metricstest.go

```diff
@@ -195,28 +198,32 @@ func lastRow(t test.T, name string, meter view.Meter) *view.Row {
 }
 
 func checkExactlyOneRow(t test.T, name string) (*view.Row, error) {
+	rows, err := readRowsFromAllMeters(name)
+	if err != nil || len(rows) == 0 {
+		return nil, fmt.Errorf("could not find row for %q", name)
+	}
+	if len(rows) > 1 {
+		return nil, fmt.Errorf("expected 1 row for metric %q got %d", name, len(rows))
+	}
+	return rows[0], nil
+}
+
+func readRowsFromAllMeters(name string) ([]*view.Row, error) {
 	// view.Meter implements (and is exposed by) metricproducer.GetAll. Since
 	// this is a test, reach around and cast these to view.Meter.
-	var retval *view.Row
+	var rows []*view.Row
 	for _, producer := range metricproducer.GlobalManager().GetAll() {
 		meter := producer.(view.Meter)
-
 		d, err := meter.RetrieveData(name)
 		if err != nil || len(d) == 0 {
 			continue
 		}
-		if len(d) > 1 {
-			return nil, fmt.Errorf("expected 1 row for metric %q got %d", name, len(d))
-		}
-		if retval != nil {
-			return nil, fmt.Errorf("got 2 rows from different meters: %+v, %+v", *retval, d[0])
+		if rows != nil {
```

Comment:
it might be useful to store the initial `meter` as well

Reply:
If we find this is having a problem, the `meter` pointers probably won't help that much.

Better would be to use the [`Read` method from `metricproducer.Producer`](https://godoc.org/go.opencensus.io/metric/metricproducer#Producer), but that has a different signature, so it doesn't help for current code.

Reply:
I'm planning to put together a PR to introduce a new set of interfaces to `metricstest` that will read out the metrics from `metricproducer` instead of the `RetrieveData` method.